cider/exec/module/CiderRuntimeModule.cpp

```diff
@@ -744,14 +744,32 @@ CiderBatch CiderRuntimeModule::setSchemaAndUpdateCountDistinctResIfNeed(
       int flatten_index = schema->getFlattenColIndex(i);
       if (schema->getColHints()[i] == ColumnHint::PartialAVG &&
           target_infos[flatten_index].agg_kind == SQLAgg::kSUM &&
-          target_infos[flatten_index].agg_arg_type.is_integer() &&
           generator::getSQLTypeInfo(schema->getColumnTypeById(i).struct_().types(0))
```

Comment:
1. `schema->getColumnTypeById(i).struct_().types(0).has_fp64()` would be better?
2. what if this is not a double column?

Reply:
and for L737, `may have` or `must have`?

Reply:
And why do we do such process in this function `setSchemaAndUpdateCountDistinctResIfNeed`? I think it's confusing, we need do some refactor later.

Reply:
> 1. `schema->getColumnTypeById(i).struct_().types(0).has_fp64()` would be better?
should be okay.
> 2. what if this is not a double column?
In this specific partial avg case, actual for velox, we need to convert bigint to double to match the output schema. In other case, like normal `sum`, the data should already be correct after extract, so we don't need special handle.